src/managers/tls.py

```diff
@@ -121,3 +133,73 @@ def delete_certificates(self, cert_type: TLSType) -> None:
             self.workload.remove_file(self.workload.paths.tls.peer_ca)
             self.workload.remove_file(self.workload.paths.tls.peer_key)
         logger.debug(f"Deleted {cert_type.value} certificate")
+
+    def clean_cas(self, tls_type: TLSType) -> None:
+        """Clean the CAs from the system.
+
+        Args:
+            tls_type (TLSType): The TLS type.
+        """
+        cas = self._load_trusted_ca(tls_type)
+        # clear cas file
+        cas_path = (
+            self.workload.paths.tls.client_ca
+            if tls_type == TLSType.CLIENT
+            else self.workload.paths.tls.peer_ca
+        )
+        self.workload.remove_file(cas_path)
+        # The last CA is the new CA, so we add it back
+        self.add_trusted_ca(cas[-1], tls_type)
+
+    def set_ca_rotation_state(self, tls_type: TLSType, state: TLSCARotationState) -> None:
+        """Set the CA rotation state.
+
+        Args:
+            tls_type (TLSType): The TLS type.
+            state (TLSCARotationState): The CA rotation state.
+        """
+        logger.debug(f"Setting {tls_type.value} CA rotation state to {state}")
+        self.state.unit_server.update({f"tls_{tls_type.value}_ca_rotation": str(state.value)})
+
+    def is_new_ca_saved_on_all_servers(self, cert_type: TLSType) -> bool:
+        """Check if the new CA is saved on all servers.
+
+        Args:
+            cert_type (TLSType): The certificate type.
+        """
+        is_ca_stored_on_all_servers = True
+        for server in self.state.servers:
+            server_ca_rotation_state = (
+                server.tls_peer_ca_rotation_state
+                if cert_type == TLSType.PEER
+                else server.tls_client_ca_rotation_state
+            )
+            if server_ca_rotation_state in [
+                TLSCARotationState.NO_ROTATION,
+                TLSCARotationState.NEW_CA_DETECTED,
+            ]:
+                is_ca_stored_on_all_servers = False
+                break
+        return is_ca_stored_on_all_servers
```

Comment:
nit - can be simplified and remove the variable:
```suggestion
                return False
                
        return True
```